src/preamble.js

```diff
@@ -1224,7 +1218,14 @@ if (Module['buffer']) {
     assert(TOTAL_MEMORY % WASM_PAGE_SIZE === 0);
 #endif
 #if ALLOW_MEMORY_GROWTH
+#if BINARYEN_MEM_MAX
+#if ASSERTIONS
+    assert({{{ BINARYEN_MEM_MAX }}} % WASM_PAGE_SIZE == 0);
```

Comment:
we might as well check that at compile time in emcc.py alongside the other checks?